src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/DumplingHelper.py

```diff
@@ -16,10 +17,19 @@ def install_dumpling():
       url = "https://dumpling.azurewebsites.net/api/client/dumpling.py"
       scriptPath = os.path.dirname(os.path.realpath(__file__))
       downloadLocation = scriptPath + "/dumpling.py"
-      urllib.urlretrieve(url, downloadLocation)
-      subprocess.call([sys.executable, downloadLocation, "install", "--update"])
+      response = urllib2.urlopen(url)
+      if response.getcode() == 200:
+        with open(downloadLocation, 'w') as f:
+          f.write(response.read())
+        subprocess.call([sys.executable, downloadLocation, "install", "--update"])
+      else:
+        raise urllib2.URLError("HTTP Status Code" + str(result.getcode()))
 
     subprocess.call([sys.executable, dumplingPath, "install"])
+  except urllib2.HTTPError, e:
+    print("Dumpling cannot be installed due to " + e.reason + " and HTTP Status Code " + str(e.code))
+  except  urllib2.URLError, e:
```

Comment:
```python
except (urllib2.HTTPError, urllib2.URLError) as e:
   print("Dumpling cannot be installed due to " + e.reason + " and HTTP Status Code " + str(e.code))
```

Reply:
I suggest "Dumpling cannot be retrieved from " + url + "  due to " + e.reason + " and HTTP Status Code " + str(e.code))

Reply:
URLError doesnot returns e.code